homeassistant/components/escea/climate.py

```diff
@@ -0,0 +1,299 @@
+            """Disconnected from controller."""
+            if ctrl is not self._controller:
+                return
+            self.set_available(False, ex)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_DISCONNECTED, controller_disconnected
+            )
+        )
+
+        @callback
+        def controller_reconnected(ctrl: Controller) -> None:
+            """Reconnected to controller."""
+            if ctrl is not self._controller:
+                return
+            self.set_available(True)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_RECONNECTED, controller_reconnected
+            )
+        )
+
+        @callback
+        def controller_update(ctrl: Controller) -> None:
+            """Handle controller data updates."""
+            if ctrl is not self._controller:
+                return
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_UPDATE, controller_update
+            )
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @callback
+    def set_available(self, available: bool, ex: Exception = None) -> None:
+        """Set availability for the controller."""
+        if self.available == available:
+            return
+
+        if available:
+            _LOGGER.info("Reconnected controller %s ", self._controller.device_uid)
+        else:
+            _LOGGER.info(
+                "Controller %s disconnected due to exception: %s",
+                self._controller.device_uid,
+                ex,
+            )
+
+        self._available = available
+        self.async_write_ha_state()
+
+    @property
+    def device_info(self):
+        """Return the device info for the Escea system."""
+        return self._device_info
+
+    @property
+    def unique_id(self):
+        """Return the ID of the controller device."""
+        return self._controller.device_uid
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return f"Escea Fireplace {self._controller.device_uid}"
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return self._supported_features
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def precision(self) -> float:
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
```

Comment:
These could be define using `_attr_*` class attributes to avoid boilerplate property getters. See https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/__init__.py#L181 for list of available attributes.

Reply:
Thanks, this was a great tip! I cleaned up all I could, plus found a few more from the base entity.